### PR TITLE
[RA2 Ch02] - restructure of RA2 chapters

### LINKED ISSUE #1018: RA2 Kubernetes Requirement Groups
Some requirements are a subset of other requirements. I think it would make sense to break these requirements into sections a bit more clearly and into sections. E.g.

# Global Requirements (G)
* req G1: Kubernetes should support persistent storage as defined in "PS"

# Persistent Storage Requirement (PS)
If req A is present, it must/should support:
* PS1: Persistent storage exposing *block storage* should use CSI
* PS2: Persistent storage exposing *file storage* should use CSI
* PS3: Object storage service may be available

### LINKED ISSUE #1112: [RA2 Ch2] - review CNI and other networking requirements
Question, why is a CNI multiplexer being made a hard requirement as opposed to just setting a requirement to support multiple network interfaces?

> "Because req.inf.ntw.01 requires the architecture to support CNI and req.inf.ntw.16 requires the capability to attach several network interfaces to the pods the architecture must support a CNI metaplugin/CNI multiplexer."

### LINKED ISSUE #1137: [RA2 Ch2] Improve traceability from RA2 to RM
references, values, etc.

### LINKED ISSUE #1338: [RA2 Ch02-04]: Separate requirements, functional capabilities and solutions
RA2 meeting on 2020.03.19 agreed to implement @tomkivlin -s suggestion from [issue 1112](https://github.com/cntt-n/CNTT/issues/1112#issuecomment-601162790):
- Ch02 - document requirements, avoid those that define a specific choice
- Ch03 - describe functional capabilities and interfaces - maybe with examples such as CNI/NSM but avoid if possible
- Ch04 - articulate the specifications - i.e. the specific choices and configurations that deliver the capabilities/interfaces (ch03) and the requirements (ch02)

Which should mean CNI is removed from ch02, the networking model is described in ch03 and then the specific interfaces (CNI, eBPF, etc.) and networking solutions (Calico, DANM, Multus, NSM) can be described in ch04. This may also mean ch05 is not required as the interface specs are all in ch04.

### PR DESCRIPTION
**Pull request by tomkivlin**
Resolves #1018 
Resolves #1112 
Resolves #1137 
Resolves #1338

### CODE REVIEW

**Comment by tomkivlin** [comment]
Changes made to chapter02.md:

Modified how RM requirements are pulled through.
Added security reqs from RM ch07.
Added  req.gen.cnt.04
Removed req.inf.com.03 - let's define CRI in ch04 in support of new req.gen.cnt.04
Removed req.inf.stg.01 - let's define how storage is provided in ch04
Removed req.inf.stg.02 - object storage covered in RM requirements
Removed req.inf.stg.03 - let's define CSI in ch04 in support of new req.gen.cnt.04
Removed req.inf.stg.04 - let's define how storage is provided in ch04
Removed req.inf.stg.05 - storage non-functionals covered in RM requirements
Removed req.inf.stg.06 - ephemeral storage covered in RM requirements
Removed req.inf.stg.07 - persistent storage covered in RM requirements
Removed req.inf.ntw.01 - let's define CNI in ch04 in support of new req.gen.cnt.04
Removed req.inf.ntw.02 - covered by what is now req.inf.ntw.08
Removed req.inf.ntw.03 - covered by what is now req.inf.ntw.08
Removed req.inf.ntw.04 - covered by RM reqs
Moved req.inf.ntw.07 to General section so applies to all areas
Moved req.inf.ntw.10 to General section and re-worded to cover all areas
Removed req.inf.ntw.15 - let's define SMI in ch04 in support of new req.gen.cnt.04
Removed req.inf.ntw.16 - covered by RM reqs
Removed req.inf.ntw.22 - let's deal with "how" in ch04 specs
Removed req.inf.acc.01 - covered by RM reqs
Removed req.inf.acc.02 - covered by RM reqs
Removed commented req.inf.ntw.05 - covered by RM reqs
Removed commented req.inf.vir.01 - new version already in main table
Removed commented req.kcm.02 - let's include in ch04 as a specification to support cloud native reqs
Removed commented req.kcm.04 - not in scope
Removed commented req.kcm.05 - covered by RM reqs
Removed commented req.kcm.08 - covered by req.gen.rsl.01
Removed commented req.kcm.09 - covered by RM reqs
Removed req.int.api.03 - let's include in ch04 in support of other existing reqs
Removed req.asr.mon.01 - covered by RM reqs
Removed req.asr.mon.02 - partially covered by RM reqs, and also let's define "how" in ch04
Removed req.asr.mon.03 - covered by RM reqs
Removed req.asr.mon.04 - covered by RM reqs
Removed req.sec.*.* - all covered by RM reqs

Other typos and renumbering

**Comment by tomkivlin** [comment]
Ready for review- propose that chapter 04 updates are done post-Baldy to resolve #1525 and other ch04-related backlog items.

**Comment by tomkivlin** [comment]
Also removed the chapters as agreed, and renamed security chapter to Security Guidance.

File: doc/ref_arch/kubernetes/chapters/chapter02.md [hunk 1]
```diff
@@ -8,306 +8,309 @@
...
+| [7.11.7](../../../ref_model/chapters/chapter07.md#7117-monitoring-and-security-audit) | sec.mon.015 | The Platform must ensure that the Monitoring systems are never starved of resources |  |
+| [7.11.7](../../../ref_model/chapters/chapter07.md#7117-monitoring-and-security-audit) | sec.mon.016 | The Platform Monitoring components should follow security best practices for auditing, including secure logging and tracing |  |
+| [7.11.7](../../../ref_model/chapters/chapter07.md#7117-monitoring-and-security-audit) | sec.lcm.017 | The Platform must Audit systems for any missing security patches and take appropriate actions |  |
+| [7.11.8](../../../ref_model/chapters/chapter07.md#7118-compliance-with-standards) | sec.std.001 | The Cloud Operator **should** comply with Center for Internet Security CIS Controls ([https://www.cisecurity.org/](https://www.cisecurity.org/)) | |
+| [7.11.8](../../../ref_model/chapters/chapter07.md#7118-compliance-with-standards) | | [Q: Are we going to verify compliance w Controls? If not, then why a “must” – but making it a “should” implies only guidance and not a control.] |  |
```
Comment (CsatariGergely) [inline doc/ref_arch/kubernetes/chapters/chapter02.md:257]: I think this is not a requirement.

Comment (tomkivlin) [reply doc/ref_arch/kubernetes/chapters/chapter02.md]: Good spot - removed in https://github.com/cntt-n/CNTT/pull/1524/commits/20414d04b5fd753639eac54223ddb12649947d7e.

**Comment by tomkivlin** [comment]
> What do you think?

I reckon start from master, if you're only updated chapter03.md and chapter04.md - I didn't make any/may updates to those so any conflicts in those files will be small, if at all.

**Comment by CsatariGergely** [comment]
> 
> 
> > What do you think?
> 
> I reckon start from master, if you're only updated chapter03.md and chapter04.md - I didn't make any/may updates to those so any conflicts in those files will be small, if at all.

Okey, I will start two pr-s to cover those changes from master and based on this pr.

File: doc/ref_arch/kubernetes/chapters/chapter05.md [hunk 2]
```diff
@@ -1,39 +1,171 @@
...
-> Compute, storage, networking, etc. that is used by the VNF/CNF workloads
 
-<a name="5.4"></a>
-## 5.4 Internal interfaces
+Thus the primary isolation mechanism in Kubernetes environment should be VM or physical machine based isolation. This means: multiple container applications should not be deployed together in the same Kubernetes cluster - unless those have been planned and verified to co-exist.
 
-> Compute, storage, networking, etc. that is used by the Kubernetes platform itself (i.e. by worker nodes)
+### Container Isolation in Kubernetes Cluster
+#### Namespaces  
+Kubernetes namespaces should be used to provide resource isolation within a Kubernetes cluster. Kubernetes should be used to isolate different types of workloads like Development, Production or Test. The default is to allocate one namespace per Containerised Network Function (CNF).
```
Comment (yu3peng) [inline doc/ref_arch/kubernetes/chapters/chapter05.md:171]: Is "Cloud Native Network Function(CNF)" better than "Containerised Network Function (CNF)"?

**Pull request merged**